test/datasets/test_dbpedia.py

```diff
@@ -0,0 +1,85 @@
+import os
+import random
+import string
+import tarfile
+from collections import defaultdict
+from unittest.mock import patch
+
+from parameterized import parameterized
+from torchtext.datasets.dbpedia import DBpedia
+
+from ..common.case_utils import TempDirMixin, zip_equal
+from ..common.torchtext_test_case import TorchtextTestCase
+
+
+def _get_mock_dataset(root_dir):
+    """
+    root_dir: directory to the mocked dataset
+    """
+    base_dir = os.path.join(root_dir, "DBpedia")
+    temp_dataset_dir = os.path.join(base_dir, "dbpedia_csv")
```

Comment:
Can you name this `temp_dataset_dir`. If the temp folder name is the same as the folder name within the tar, the tar extraction will be skipped. This is because the dataset implementation has some caching logic that checks whether the extracted files exists before going ahead with the tar archive extraction (this was actually missing for the DBpedia dataset but is being added here https://github.com/pytorch/text/pull/1571)